Tools/autotest/common.py

```diff
@@ -604,26 +604,53 @@ def disarm_motors_with_switch(self, switch_chan, timeout=20):
             if not self.mav.motors_armed():
                 self.progress("MOTORS DISARMED OK WITH SWITCH")
                 return True
-        self.progress("FAILED TO DISARM WITH SWITCH")
+        self.progress("Unable to DISARMED with switch")
         return False
 
-    def autodisarm_motors(self):
-        """Autodisarm motors."""
-        self.progress("Autodisarming motors")
-        if self.mav.mav_type == mavutil.mavlink.MAV_TYPE_GROUND_ROVER:  # NOT IMPLEMENTED ON ROVER
+    def wait_autodisarm_motors(self):
+        """Wait Autodisarm motors."""
```

Comment:
This comment needs some help :-)

Also, this is most certainly not waiting for the vehicle to auto-disarm; it is actively disarming it, so it is now misnamed / misfeatured.

Reply:
Better fix  may be to override in a subclass for any autodisarm test.